src/legacy/core_plugins/metrics/common/get_last_value.js

```diff
@@ -17,24 +17,20 @@
  * under the License.
  */
 
-import _ from 'lodash';
-export default (data, lookback = 1) => {
-  if (_.isNumber(data)) return data;
-  if (!Array.isArray(data)) return 0;
-  // First try the last value
-  const last = data[data.length - 1];
-  const lastValue = Array.isArray(last) && last[1];
-  if (lastValue) return lastValue;
+import { isArray, findLast } from 'lodash';
 
-  // If the last value is zero or null because of a partial bucket or
-  // some kind of timeshift weirdness we will show the second to last.
-  let lookbackCounter = 1;
-  let value;
-  while (lookback > lookbackCounter && !value) {
-    const next = data[data.length - ++lookbackCounter];
-    value =  _.isArray(next) && next[1] || 0;
+const DEFAULT_VALUE = 0;
+
+export default (data, defaultValue = DEFAULT_VALUE) => {
+  if (!isArray(data)) {
+    return data;
   }
-  return value || 0;
-};
 
+  const extractValue = data => data && data[1] || null;
 
+  // If the last value is zero or null because of a partial bucket or
+  // some kind of timeshift weirdness we will show the second to last.
+  const lastValid = findLast(data, item => extractValue(item));
```

Comment:
Could be simplified, not necessary, just a nitpick
```suggestion
  const lastValid = findLast(data, extractValue);
```

Reply:
not sure that it's better